### DISCUSSION TITLE
Support import from other Authenticator apps

### DISCUSSION THREAD

**Discussion by vishnukvmd** [ente-io/ente#481] [Enhancements]
- [x] [Raivo](https://github.com/raivo-otp/ios-application)
- [x] [Aegis](https://github.com/beemdevelopment/Aegis/)
- [x] Google Authenticator
- [x] 2FAS
- [x] Bitwarden
- [x] Lastpass
- [ ] [Authenticator Pro](https://github.com/jamie-mh/AuthenticatorPro)
- [ ] [andOTP](https://github.com/andOTP/andOTP)
- [x] [OTPAuth](https://apps.apple.com/us/app/otp-auth/id659877384)

**Comment by ThatOneCalculator** [comment]
And maybe encrypted backups from [Authenticator Pro](https://github.com/jamie-mh/AuthenticatorPro) would be nice

**Comment by asdfzxcvbn** [comment]
Encrypted backups from [Raivo](https://github.com/raivo-otp/ios-application) would also be nice!

**Comment by muznyo** [comment]
also from Bitwarden

**Comment by ThatOneCalculator** [comment]
Bitwarden bundles passwords and TOTP codes so I don't think Bitwarden import would be that viable

**Comment by floschoepfer** [comment]
What about Twilio Authy?

**Reply by Damariobros** [reply]
I don't know how viable that is. Until recently, there was but one way to extract secrets from your Authy account, and that was to install an old desktop version with a debug port, open the debug port using command prompt, access that debug port in a Chromium browser, and use a script in the Inspect Element console to extract them. Then manually copy the secrets or scan each individual QR code produced in output.

With the shutdown of Authy Desktop, they killswitched it from the server end AND the client end, so that all attempted logins from desktop are rejected, all installations are killed, and all desktops are removed from people's accounts. I doubt user agent spoofing would be able to get around this.

I don't think there is a way anymore.

Screw Authy.

**Comment by darthShadow** [comment]
+1 for Twilio Authy.

**Comment by LearningAsIGo71** [comment]
Another vote for Authy

**Comment by jerryn70** [comment]
Authy is needed

**Comment by K0-RR** [comment]
Also KeePassXC database. That's what I currently use for OTP on both Desktop and Android.

**Comment by ThiefMaster** [comment]
+1 for andOTP

**Reply by UC-VR** [reply]
Still none...

**Reply by joicemjoseph** [reply]
Raised one merge request. I wanted to migrate from andOTP, and added support for andotp. please give it a review: https://github.com/ente-io/ente/pull/8179

**Comment by t-moe** [comment]
FreeOTP

**Comment by Caian** [comment]
Even though Raivo is already supported, it would be nice to import it directly from the encrypted zip to avoid the raw json touching the filesystem.

**Comment by halshar** [comment]
andOTP please

**Reply by joicemjoseph** [reply]
[added](https://github.com/ente-io/ente/pull/8179)

**Comment by fessmm** [comment]
Any news on [Authenticator Pro](https://github.com/jamie-mh/AuthenticatorPro) ?

**Comment by marianstiehler** [comment]
Hey everyone,

I would love to use ente authenticator, but I have much too many authenticator codes in 2Stable's authenticator app. It's one of the most downloaded authenticator apps on the Apple App Store (but it's macOS/iOS only).

It does let you export your codes into JSON: (redacted for security)

"items":[{"primary":"2887E395-A274-5673-8DC0-C2E1B923BD9B","icon":"google.com","account":"test@googlemail.com","issuer":"Google","secret":"<OTP secret>","period":30,"notes":"<notes>","added":"2022-12-29T06:12:39.106045961Z","isExcludedFromWatch":true,"hosts":["google.com"]},…

Is there any chance someone could write an import script?

I will look into writing one myself, but I am inexperienced.

More information: https://authenticator.2stable.com/
